src/test/daml/Daml/Finance/Asset/Test/IbericoPig.daml

```diff
@@ -0,0 +1,147 @@
+      --                         Aged -> [Restaurant]
+      --                         Restaurant -> [Restaurant]
+      --                         _ -> [] in
+      --   map getClaimFromState remainingStates
+      nextState: State
+      nextState = 
+        case state of 
+          Pig -> Butchered
+          Butchered -> Salted
+          Salted -> Aged
+          Aged -> Restaurant
+          Restaurant -> Eaten
+          Eaten -> Eaten -- way to abort and remove this
+    
+    implements HasClaims.I where
+      view = HasClaims.View with acquisitionTime = (toUTCTime . Unit) matureDate
+      getClaims = do
+        prepareAndTagClaims [getClaimFromState state] (show state)
+
+    implements Instrument.I where 
+      asDisclosure = toInterface @Disclosure.I this
+      view = Instrument.View with issuer = farmer; depository = farmer; id; validAsOf = (toUTCTime . Unit) matureDate
+      getKey = instrumentKey
+
+    implements Lifecyclable.I where
+      view = Lifecyclable.View with lifecycler = farmer -- should not be farmer
+      lifecycle Lifecyclable.Lifecycle{ruleName; settler; eventCid; clockCid; observableCids} self = do
+        t <- Event.getEventTime <$> fetch eventCid
+        let 
+          claimPig = toInterface @HasClaims.I this
+        (remaining, pending) <- Lifecycle.lifecycle observableCids claimPig [Lifecycle.timeEvent t]
+        let 
+         (consumed, produced) = Lifecycle.splitPending pending
+         newKey = instrumentKey with id.version = sha256 $ show remaining
+        Some event <- fromInterface @StateTransitionEvent <$> fetch eventCid
+        newInstrumentCid <- create this with id = newKey.id; state = nextState
+        let 
+          settlementDate = toDateUTC event.eventTime
+        effectCid <- toInterfaceContractId <$> create Effect with 
+          provider = farmer
+          settler
+          targetInstrument = instrumentKey
+          producedInstrument = Some newKey
+          consumed
+          produced
+          settlementDate
+          id = id.label <> "-" <> show settlementDate
+          observers = (.observers) . _view $ toInterface @Disclosure.I this
+        pure (toInterfaceContractId newInstrumentCid, [effectCid])
+
+    implements Disclosure.I where
+      view = Disclosure.View with disclosureControllers = singleton $ singleton farmer; observers
+      setObservers Disclosure.SetObservers{newObservers} = do
+        cid <- toInterfaceContractId <$> create this with observers = newObservers
+        Instrument.disclosureUpdateReference newObservers instrumentKey cid
+      archive' self = archive (coerceContractId self : ContractId IbericoPig)
+
+
+prepareAndTagClaims : Applicative f => [Claim Date Decimal Deliverable Text] -> Text -> f [TaggedClaim]
+prepareAndTagClaims claim tag = do
+  let
+    claims = mapClaimToUTCTime $ mconcat claim
+  pure [taggedClaim tag claims]
+
+
+mapClaimToUTCTime : Claim Date Decimal Deliverable Text -> C
```

Comment:
I cannot avoid this redundant conversion, even when I don't really need time constraint on claim

Reply:
You can get rid of that conversion by changing the signature of `getClaimFromState` from `State -> Claim Date Decimal Deliverable Text` to `State -> Claim Time Decimal Deliverable Text`.

There shouldn't be any other change required given that time is currently not being used.

Reply:
Oh, yes you are totally right! thanks for the tip